react-components/src/components/Reveal3DResources/Reveal3DResources.tsx

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { useRef, type ReactElement, useContext, useState, useEffect } from 'react';
+import { type Cognite3DViewer } from '@cognite/reveal';
+import { ModelsLoadingStateContext } from './ModelsLoadingContext';
+import CadModelContainer from '../CadModelContainer/CadModelContainer';
+import PointCloudContainer from '../PointCloudContainer/PointCloudContainer';
+import { Image360CollectionContainer } from '../Image360CollectionContainer/Image360CollectionContainer';
+import { useReveal } from '../RevealContainer/RevealContext';
+import {
+  type AddReveal3DModelOptions,
+  type AddImageCollection360Options,
+  type TypedReveal3DModel,
+  type AddResourceOptions
+} from './types';
+
+export type Reveal3DResourcesProps = {
+  resources: AddResourceOptions[];
+};
+
+export const Reveal3DResources = ({ resources }: Reveal3DResourcesProps): ReactElement => {
+  const [reveal3DModels, setReveal3DModels] = useState<TypedReveal3DModel[]>([]);
+  const { setModelsAdded } = useContext(ModelsLoadingStateContext);
+  const viewer = useReveal();
+  const numModelsLoaded = useRef(0);
+
+  useEffect(() => {
+    getTypedModels(resources, viewer).then(setReveal3DModels).catch(console.error);
+  }, []);
+
+  const image360CollectionAddOptions = resources.filter(
+    (resource): resource is AddImageCollection360Options =>
+      (resource as AddImageCollection360Options).siteId !== undefined
+  );
+
+  const onModelLoaded = (): void => {
+    numModelsLoaded.current += 1;
+    if (numModelsLoaded.current === resources.length) {
+      setModelsAdded(true);
+    }
+  };
+
+  return (
+    <>
+      {reveal3DModels
+        .filter(({ type }) => type === 'cad')
+        .map((addModelOption) => {
+          return (
+            <CadModelContainer
+              key={`${addModelOption.modelId}/${addModelOption.revisionId}`}
```

Comment:
In case someone for some weird reason wants to add two exactly the same models the following error will appear. 
![image](https://github.com/cognitedata/reveal/assets/45129444/834ee411-0d29-45be-b055-cb5fdcee83bb)
I think we should add some salt/uuid/model number to the key here and in pointcloud/360 image case.